test/helpers/kubectl.go

```diff
@@ -726,6 +746,82 @@ func (kub *Kubectl) GetCiliumPodOnNode(namespace string, node string) (string, e
 	return res.Output().String(), nil
 }
 
+func (kub *Kubectl) DeployCiliumDS(ciliumOpts *CiliumOpts) error {
+	if config.CiliumTestConfig.Developer {
+		s := InitRuntimeHelper(K8s1VMName(), kub.logger)
+		err := s.SetUpCiliumK8s(ciliumOpts)
+		if err != nil {
+			return err
+		}
+		s = InitRuntimeHelper(K8s2VMName(), kub.logger)
+		return s.SetUpCiliumK8s(ciliumOpts)
+	}
+
+	var path string
+	if ciliumOpts != nil && ciliumOpts.TunnelType == "geneve" {
+		path = kub.ManifestGet("cilium_ds_geneve.yaml")
+	} else {
+		path = kub.ManifestGet("cilium_ds.yaml")
+	}
+
+	kub.Apply(path)
+	_, err := kub.WaitforPods(KubeSystemNamespace, "-l k8s-app=cilium", 600)
+	return err
+}
+
+func (kub *Kubectl) WaitToDeleteCilium(logger *logrus.Entry) {
+	status := 1
+	for status > 0 {
+		pods, err := kub.GetCiliumPods(KubeSystemNamespace)
+		status := len(pods)
+		logger.Infof("Cilium pods terminating '%d' err='%v' pods='%v'", status, err, pods)
+		if status == 0 {
+			return
+		}
+		time.Sleep(1 * time.Second)
+	}
+}
+
+func (kub *Kubectl) DeleteCiliumDS(ciliumOpts *CiliumOpts) error {
+	if config.CiliumTestConfig.Developer {
+		return nil
+	}
+	var path string
+	if ciliumOpts != nil && ciliumOpts.TunnelType == "geneve" {
+		path = kub.ManifestGet("cilium_ds_geneve.yaml")
+	} else {
+		path = kub.ManifestGet("cilium_ds.yaml")
+	}
+	kub.Delete(path)
+	kub.WaitToDeleteCilium(kub.logger)
+	return nil
+}
+
+func (kub *Kubectl) GetCilium(node string) (pod, ip string, err error) {
+	if config.CiliumTestConfig.Developer {
+		switch node {
+		case K8s1:
+			return node, "192.168.33.11", nil
```

Comment:
I am not a fan of hardcoding stuff like this across the code. It makes the Ginkgo framework very reliant upon the implementation in the Vagrantfile, which is already tightly coupled enough as it is.